src/app/github-name-to-id.pipe.ts

```diff
@@ -0,0 +1,67 @@
+import { HttpBackend, HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { Pipe, PipeTransform } from '@angular/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { SessionQuery } from './shared/session/session.query';
+import { TokenQuery } from './shared/state/token.query';
+import { UserQuery } from './shared/user/user.query';
+
+/**
+ * This converts an organization name or username to a GitHub apps installation link
+ * Works for a user's name and a GitHub organization returned by the /user/orgs GitHub endpoint
+ *
+ * @export
+ * @class GithubNameToIdPipe
+ * @implements {PipeTransform}
+ */
+@Pipe({
+  name: 'githubNameToId'
+})
+export class GithubNameToIdPipe implements PipeTransform {
+  constructor(private tokenQuery: TokenQuery, private httpBackend: HttpBackend, private userQuery: UserQuery,
+    private sessionQuery: SessionQuery) {}
+  transform(userNameOrOrganizationName: string): Observable<string> {
+    const username = this.userQuery.getSnapshot().user.name;
+    if (userNameOrOrganizationName === username) {
+      return this.getIdFromUsername(username);
+    }
+    return this.tokenQuery.gitHubOrganizations$.pipe(
+      map((organizationsResponse: Array<any>) => {
```

Comment:
You *could* define an interface for the gitHubOrganizations, at least covering the field(s) you care about.